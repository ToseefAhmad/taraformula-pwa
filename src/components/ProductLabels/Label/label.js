import PropTypes, { shape, string } from 'prop-types';
import React, { useMemo } from 'react';

import { toCamelCase } from '../utils';

import resourceUrl from '@magento/peregrine/lib/util/makeUrl';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import Image from '@magento/venia-ui/lib/components/Image';
import RichContent from '@magento/venia-ui/lib/components/RichContent';

import defaultClasses from './label.module.css';

/**
 * Cut html tags from string
 *
 * @return {String}.
 */
const stripText = value => {
    return value.replace(/(<([^>]+)>)/gi, '');
};

/**
 * @typedef Label
 * @kind functional component
 *
 * @param {props} props React component props
 *
 * @returns {React.Element}
 */
const Label = props => {
    const { item, dimensions, settingStyle, specialWidth } = props;
    const classes = mergeClasses(defaultClasses, props.classes);

    const itemStyle = useMemo(() => {
        const css = item.style;
        const result = {};
        const rules = css.split(';');

        if (Array.isArray(rules) && rules.length) {
            rules.forEach(rule => {
                let [key, value] = rule.split(':');

                if (key && value) {
                    key = toCamelCase(key.trim());
                    value = value.trim();
                    result[key] = value;
                }
            });
        }

        return result;
    }, [item]);

    /**
     * Set Label width in percent of product image
     *
     * @return {Number}.
     */
    const scaledWidth = useMemo(() => {
        const dimensionToUse = specialWidth === undefined ? dimensions.width : specialWidth;
        return item.size ? (dimensionToUse / 3) * (parseInt(item.size) / 100) : dimensionToUse / 3;
    }, [dimensions.width, item.size, specialWidth]);

    /**
     * Set responsive font-size and max-height for Label text
     *
     * @return {Object}.
     */
    const getTextStyles = () => {
        return {
            fontSize: (scaledWidth < 100 ? scaledWidth : 100) + '%',
            maxWidth: scaledWidth
        };
    };

    /**
     * Combine css styles: static from backend and dynamic width while resize
     *
     * @return {Object}.
     */
    const getLabelStyles = () => ({
        ...settingStyle,
        ...itemStyle,
        ...{ width: scaledWidth + 'px', height: scaledWidth + 'px' }
    });

    let imgSrc = resourceUrl(item.image ? `/${item.image.replace(/^\//, '')}` : '', {
        type: 'image-amasty',
        width: scaledWidth * 2,
        fit: 'contain'
    });

    imgSrc = imgSrc.substring(imgSrc.indexOf('/media'));

    /**
     * Render Label component
     *
     */
    return (
        <div style={getLabelStyles()} className={classes.root}>
            {item.image && (
                <Image
                    alt={stripText(item.txt)}
                    classes={{
                        image: classes.image,
                        root: classes.imageContainer
                    }}
                    displayPlaceholder={false}
                    src={imgSrc}
                    width={scaledWidth}
                />
            )}

            <div className={classes.text} style={getTextStyles()}>
                <RichContent html={item.txt} />
            </div>
        </div>
    );
};

/**
 * Props for {@link Label}
 *
 * @typedef props
 *
 * @property {Object} classes An object containing the class names for the AmLabel component.
 */
Label.propTypes = {
    classes: shape({
        root: string,
        image: string,
        imageContainer: string,
        text: string,
        topLeft: string,
        topCenter: string,
        topRight: string,
        middleLeft: string,
        middleCenter: string,
        middleRight: string,
        bottomLeft: string,
        bottomCenter: string,
        bottomRight: string
    }),
    item: PropTypes.object,
    dimensions: shape({
        width: PropTypes.number
    }),
    settingStyle: PropTypes.any,
    specialWidth: PropTypes.number
};

export default Label;
