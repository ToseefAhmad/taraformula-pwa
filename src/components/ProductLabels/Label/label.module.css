.root {
    position: relative;
    z-index: 1;
    display: flex;
    font-size: 1em;
    height: 100%;
    width: 100%;
}

.image {
    position: static;
    width: 100%;
    max-height: 100%;
    height: auto;
    visibility: visible;
}

.image:first-of-type {
    display: none;
}

.imageContainer {
    position: relative;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    display: flex;
    justify-content: center;
    opacity: 1;
    max-height: 100%;
    width: 100%;
    height: 100%;
}

.text {
    width: 100%;
    font-size: 100%;
    line-height: 1.2;
    text-align: inherit;
}

.imageContainer + .text {
    position: absolute;
    top: 50%;
    left: 0;
    overflow: hidden;
    text-align: center;
    transform: translateY(-50%);
}

.imageContainer .image {
    width: 100% !important;
    height: unset !important;
    position: absolute !important;
    left: 0 !important;
    top: 0 !important;
    bottom: 0 !important;
    right: 0 !important;
    margin-top: unset;
}
