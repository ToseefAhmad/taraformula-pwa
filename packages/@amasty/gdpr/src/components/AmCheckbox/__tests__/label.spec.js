import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import Label from '../label';

jest.mock('@magento/venia-ui/lib/components/LinkButton', () => 'LinkButton');
jest.mock('react-intl', () => ({
  FormattedMessage: props => (
    <div componentName="Formatted Message Component" {...props} />
  )
}));

const handleClick = jest.fn().mockName('handleClick');
const defaultProps = {
  id: 0,
  handleClick
};

test('renders default', () => {
  const props = {
    ...defaultProps,
    message: {}
  };

  const tree = createTestInstance(<Label {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders with only prevText', () => {
  const props = {
    ...defaultProps,
    message: {
      prevText: 'test'
    }
  };
  const tree = createTestInstance(<Label {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders with only text', () => {
  const props = {
    ...defaultProps,
    message: {
      text: 'test'
    }
  };
  const tree = createTestInstance(<Label {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders with only afterText', () => {
  const props = {
    ...defaultProps,
    message: {
      afterText: 'test'
    }
  };
  const tree = createTestInstance(<Label {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders with all link', () => {
  const props = {
    ...defaultProps,
    message: {
      prevText: 'test',
      text: 'test',
      afterText: 'test'
    }
  };
  const tree = createTestInstance(<Label {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
