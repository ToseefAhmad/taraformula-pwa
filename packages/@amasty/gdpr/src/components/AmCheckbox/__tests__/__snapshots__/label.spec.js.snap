// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `null`;

exports[`renders with all link 1`] = `
Array [
  <div
    componentName="Formatted Message Component"
    defaultMessage="test"
    id="privacy.prevMessage0"
  />,
  <LinkButton
    onClick={[Function]}
    type="button"
  >
    <div
      componentName="Formatted Message Component"
      defaultMessage="test"
      id="privacy.link0"
    />
  </LinkButton>,
  <div
    componentName="Formatted Message Component"
    defaultMessage="test"
    id="privacy.afterMessage0"
  />,
]
`;

exports[`renders with only afterText 1`] = `
<div
  componentName="Formatted Message Component"
  defaultMessage="test"
  id="privacy.afterMessage0"
/>
`;

exports[`renders with only prevText 1`] = `
<div
  componentName="Formatted Message Component"
  defaultMessage="test"
  id="privacy.prevMessage0"
/>
`;

exports[`renders with only text 1`] = `
<LinkButton
  onClick={[Function]}
  type="button"
>
  <div
    componentName="Formatted Message Component"
    defaultMessage="test"
    id="privacy.link0"
  />
</LinkButton>
`;
